src/extension.ts

```diff
@@ -3568,6 +3575,40 @@ export async function activate(context: vscode.ExtensionContext) {
       new HintHoverProvider(treeDataProvider)
     )
   );
+  checkAndNotifyMissingCompileCommands();
+}
+
+function checkAndNotifyMissingCompileCommands() {
+  if (vscode.workspace.workspaceFolders) {
+    vscode.workspace.workspaceFolders.forEach(async (folder) => {
+      try {
+        const isIdfProject = utils.checkIsProjectCmakeLists(folder.uri.fsPath);
+        if (isIdfProject) {
+          const compileCommandsPath = path.join(
```

Comment:
I will suggest using the build path setting instead of static build directory.

```ts
const buildDirPath = idfConf.readParameter(
        "idf.buildPath",
        workspaceRoot
      ) as string;
```